pkg/skaffold/build/cluster/secret.go

```diff
@@ -24,62 +24,69 @@ import (
 	"github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes"
 )
 
+const (
+	defaultKanikoSecretPath = "kaniko-secret"
+)
+
 func (b *Builder) setupPullSecret(out io.Writer) (func(), error) {
 	if b.PullSecret == "" && b.PullSecretName == "" {
 		return func() {}, nil
 	}
 
-	color.Default.Fprintf(out, "Creating kaniko secret [%s/%s]...\n", b.Namespace, b.PullSecretName)
-
+	color.Default.Fprintf(out, "Checking for kaniko secret [%s/%s]...\n", b.Namespace, b.PullSecretName)
 	client, err := kubernetes.Client()
 	if err != nil {
 		return nil, fmt.Errorf("getting Kubernetes client: %w", err)
 	}
 
 	secrets := client.CoreV1().Secrets(b.Namespace)
-
-	if b.PullSecret == "" {
-		logrus.Debug("No pull secret specified. Checking for one in the cluster.")
-
-		if _, err := secrets.Get(b.PullSecretName, metav1.GetOptions{}); err != nil {
-			return nil, fmt.Errorf("checking for existing kaniko secret: %w", err)
+	if _, err := secrets.Get(b.PullSecretName, metav1.GetOptions{}); err != nil {
+		color.Default.Fprintf(out, "Creating kaniko secret [%s/%s]...\n", b.Namespace, b.PullSecretName)
+		if b.PullSecret == "" {
```

Comment:
This looks like a bug to me -  why don't we fall back to default pullsecret name?

Reply:
`pullSecretName` and `pullSecret` are two different configs
1. `pullSecretName` is used to indicate the secret name. The default is `kaniko-secret` and is only set if `pullSecret` is specified.
2. `pullSecret` refers to secret file path.

With this PR, `pullSecret` is used for a dual purpose. 
1. To create a secret if it does not exist from the file specified in this value
2. To set `GOOGLE_APPLICATION_CREDENTAILS` env variable to this path when secret exists. 

A bunch of tekton users, create secrets one time using a command like below. 
See Additional Information  in linked [issue](https://github.com/GoogleContainerTools/skaffold/issues/3828)
A way to specify which path the secret lies at when using a secret is created using command if secret already exists.
```
kubectl create secret generic kaniko-secret --from-file=/Users/tejaldesai/workspace/keys/my-key.json
```

If a secret already exists, we should set the `GOOGLE_APPLICATION_CREDENTAILS` to path within the volume.

Reply:
It sounds like pullSecret should be renamed to pullSecretPath.

Reply:
done! Renamed `pullSecret` to `pullSecretPath`